stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/utils/Manufacturer.kt

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2014-2022 Stream.io Inc. All rights reserved.
+ *
+ * Licensed under the Stream License;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://github.com/GetStream/stream-chat-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getstream.chat.android.ui.utils
+
+import android.os.Build
+import io.getstream.chat.android.core.internal.InternalStreamChatApi
+
+/**
+ * Checks whether the device manufacturer is Xiaomi ot not.
+ * Fixes issue [#3255](https://github.com/GetStream/stream-chat-android/issues/3255)
+ *
+ * @return is manufacturer Xiaomi or not.
+ */
+@InternalStreamChatApi
+internal fun isXiaomi(): Boolean {
+    return Build.MANUFACTURER.lowercase() == "xiaomi"
```

Comment:
I would maybe go the other way around and make this function less specific, more abstract, something like:

```
internal fun shouldConsumeLongTap(): Boolean {
     return "xiaomi" in Build.MANUFACTURER.lowercase()
}
```

We can also pull out the xiaomi into a constant. This way we can add more cases here if there are more vendors with issues and also we don't just match it to "xiaomi" we can also match it to "xiaomi_mi" or whatever as it's a contains check

Reply:
Agreed, refactored the check so we can expand it with more manufacturers if needed.